drivers/linstorvolumemanager.py

```diff
@@ -841,31 +841,31 @@ def resize_volume(self, volume_uuid, new_size):
         self.ensure_volume_is_not_locked(volume_uuid)
         new_size = self.round_up_volume_size(new_size) // 1024
 
-        retry_count = 30
-        while True:
-            result = self._linstor.volume_dfn_modify(
-                rsc_name=volume_name,
-                volume_nr=0,
-                size=new_size
-            )
-
-            self._mark_resource_cache_as_dirty()
-
-            error_str = self._get_error_str(result)
-            if not error_str:
-                break
+        # We can't resize anything until DRBD is up to date
+        # We wait here for 5min max and raise an easy to understand error for the user
+        # This is mostly an issue for thick provisioning, thin isn't affected
+        start_time = time.monotonic()
+        try:
+            self._linstor.resource_dfn_wait_synced(volume_name, wait_interval=1.0, timeout=60*5)
```

Comment:
Where does the 5min timeout value comes from ? Have you benchmarked with different disk sizes and types ? (Not very fond of magic numbers, it's better if they can be explained or backed in comments)